### DISCUSSION TITLE
Following the Ubuntu guide results in no network support

### DISCUSSION THREAD

**Discussion by CKingX** [zbm-dev/zfsbootmenu#631] [Questions] [answered]
I followed the [Ubuntu 22.04](https://docs.zfsbootmenu.org/en/v2.3.x/guides/ubuntu/uefi.html) guide on a SATA disk encrypted ZFS and was able to install the minimal system. On boot, I logged into the root user account and attempted to install ubuntu-desktop but it could not connect to the network and timed out. Booting back into the ubuntu live, I chrooted and was able to install the ubuntu-desktop system and re-export the pool and reboot. This time, it proceeded in the GUI and I created a username. However, the network would still not work. Any guidance on this?

Thank you

**Comment by zdykstra** [comment]
You'll need to install a network manager of some kind, if ubuntu-desktop doesn't automatically depend on it. I don't run Ubuntu or have any suggestions on what that should be, though.

**Comment by CKingX** [comment]
Thank you! I was able to configure network support (by creating a file in netplan). However, I noticed I was missing things like the browser (no Firefox) and so on. I think I will likely try a different way of doing ZFS. Thank you for the help!

**Reply by Sithuk** [reply, marked as answer]
Have you tried the ubuntu zfsbootmenu install script?
https://github.com/Sithuk/ubuntu-server-zfsbootmenu

**Reply by classabbyamp** [reply]
the point of ZBM's guides is to provide users with guidance on installing a basic system that uses ZBM. Installing the applications and packages you feel are necessary is out of scope, and your distro's documentation should have that documented if needed.

**Reply by CKingX** [reply]
I am testing your script @Sithuk right now

**Reply by CKingX** [reply]
I just mean that it should support the built-in packages out of the box like a browser and networking. Networking not working out of the box can make it difficult to install packages to begin with as well (without manually making it work with netcfg).

**Reply by CKingX** [reply]
It is usable out of the box but you still need to do things like snap install Firefox (or if you prefer apt) and so on. However, it's *much* better in that you don't need to go out of your way to do things like ubuntu-desktop installed and it configured networking and so on. It even used disk by id which is a better approach.
